plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStoragePoolManager.java

```diff
@@ -138,20 +139,26 @@ public boolean connectPhysicalDisksViaVmSpec(VirtualMachineTO vmSpec) {
         List<DiskTO> disks = Arrays.asList(vmSpec.getDisks());
 
         for (DiskTO disk : disks) {
-            if (disk.getType() != Volume.Type.ISO) {
-                VolumeObjectTO vol = (VolumeObjectTO)disk.getData();
-                PrimaryDataStoreTO store = (PrimaryDataStoreTO)vol.getDataStore();
-                KVMStoragePool pool = getStoragePool(store.getPoolType(), store.getUuid());
+            if (disk.getType() == Volume.Type.ISO) {
+                result = true;
```

Comment:
Is it ok to assume this in every case?

Reply:
Yes @nvazquez, we can always assume result as true in case of an ISO; as the ISO connection does not have any error, because it is not executed. Then we continue the iteration over the VM volumes List.

If an issue happens while connecting a Disk, the result will be false as you can check at lines 157 - 161.